spring-cloud-dataflow-admin-starter/src/main/java/org/springframework/cloud/dataflow/admin/config/AdminConfiguration.java

```diff
@@ -158,9 +163,13 @@ public void configurePathMatch(PathMatchConfigurer configurer) {
 	}
 
 	@Bean
-	@ConditionalOnMissingBean
+	@ConditionalOnMissingBean(TapOnChannelRecoveryStrategy.class)
 	public RecoveryStrategy<?> tapOnChannelExpansionStrategy() {
-		return new TapOnChannelExpansionStrategy();
+		RecoveryStrategy<?> recoveryStrategy = new TapOnChannelRecoveryStrategy(streamDefinitionRepository());
+		if (this.streamCompletionProvider != null) {
```

Comment:
Ah, so [this](https://github.com/spring-cloud/spring-cloud-dataflow/commit/1317402a4280b34c20298837ae4d4e7659f20909) is the reason for it not being registered. Good catch